static/js/click_handlers.js

```diff
@@ -639,10 +639,32 @@ exports.initialize = function () {
     $(".compose_stream_button").on("click", () => {
         popovers.hide_mobile_message_buttons_popover();
         compose_actions.start("stream", {trigger: "new topic button"});
+        $("#compose-textarea").css(
+            "max-height",
+            parseInt($("#bottom_whitespace").outerHeight(), 10) -
+                (parseInt($("#below-compose-content").outerHeight(), 10) +
+                    parseInt($(".right_part").outerHeight(), 10) +
+                    parseInt($("#compose-textarea").css("padding-top"), 10) +
+                    parseInt($("#compose-textarea").css("padding-bottom"), 10) +
+                    parseInt($("#compose-textarea").css("margin-top"), 10) +
+                    parseInt($(".message_comp.compose-content").css("padding-top"), 10) +
+                    parseInt($(".message_comp.compose-content").css("padding-bottom"), 10)),
+        );
     });
     $(".compose_private_button").on("click", () => {
         popovers.hide_mobile_message_buttons_popover();
         compose_actions.start("private");
+        $("#compose-textarea").css(
+            "max-height",
+            parseInt($("#bottom_whitespace").outerHeight(), 10) -
+                (parseInt($("#below-compose-content").outerHeight(), 10) +
+                    parseInt($(".right_part").outerHeight(), 10) +
+                    parseInt($("#compose-textarea").css("padding-top"), 10) +
+                    parseInt($("#compose-textarea").css("padding-bottom"), 10) +
+                    parseInt($("#compose-textarea").css("margin-top"), 10) +
+                    parseInt($(".message_comp.compose-content").css("padding-top"), 10) +
+                    parseInt($(".message_comp.compose-content").css("padding-bottom"), 10)),
+        );
```

Comment:
The reason I pasted the code here is because , At first the <code>max-height</code> property won't be applied to <code>ComposeBox</code> since the user hasn't resized the window ( in turn , not calling the <code>exports.resize_bottom_whitespace</code> function)

Reply:
Hey , @timabbott  , Should I work on revamping the code to make it less repetitive ?

Reply:
Yeah, you should extract this big math block to a function (or perhaps better, make `compose_ui.autosize_textarea` take care of this logic and call that function?)